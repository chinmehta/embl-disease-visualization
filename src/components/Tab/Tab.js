import React from "react";

function Tab(props) {
  return <>{props.children}</>;
}

export default Tab;
