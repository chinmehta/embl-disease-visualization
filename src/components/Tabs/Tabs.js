import React, { useState } from "react";

function Tabs(props) {
  let [allTabs, setAllTabs] = useState(
    props.children.map((tab, index) => {
      const obj = {
        label: tab.props.label,
        active: index === 0 ? true : false,
        children: tab.props.children,
      };
      return obj;
    })
  );

  const toggleTabs = (newActiveIndex) => {
    setAllTabs(
      allTabs.map((tab, index) => {
        const obj = {
          label: tab.label,
          active: index === newActiveIndex ? true : false,
          children: tab.children,
        };
        return obj;
      })
    );
  };

  return (
    <React.Fragment>
      <td colSpan="4" className="border border-gray-400">
        <div className="tab-container my-2 mx-5">
          <div className="tab-list flex border-gray-400 border-b-2">
            {allTabs.map((tabElement, index) => (
              <span
                key={index}
                onClick={() => toggleTabs(index)}
                className={` px-4 cursor-pointer ${
                  tabElement.active
                    ? "bg-embl-primary-blue text-white"
                    : "bg-white text-embl-primary-blue"
                }`}
              >
                {tabElement.label}
              </span>
            ))}
          </div>
          <div>
            {allTabs.map(
              (tabElement, index) =>
                tabElement.active && (
                  <React.Fragment key={index}>
                    {tabElement.children}
                  </React.Fragment>
                )
            )}
          </div>
        </div>
      </td>
    </React.Fragment>
  );
}

export default Tabs;
