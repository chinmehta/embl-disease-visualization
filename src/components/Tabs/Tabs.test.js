import React from "react";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import Tabs from "./Tabs";
import Tab from "../Tab/Tab";

it("renders when no children is passed as prop", () => {
  const element = renderer.create(<Tabs></Tabs>).toJSON();

  expect(element).toMatchSnapshot();
});

it("renders when children is passed as prop - snapshot", () => {
  const element = renderer
    .create(
      <Tabs>
        <div label="tab text 1">Tab1</div>
        <div label="tab text 2">Tab2</div>
      </Tabs>
    )
    .toJSON();

  expect(element).toMatchSnapshot();
});

test("renders when children without tab label is passed as prop", () => {
  render(
    <table>
      <tbody>
        <tr>
          <Tabs>
            <Tab>Tab1</Tab>
            <Tab>Tab2</Tab>
          </Tabs>
        </tr>
      </tbody>
    </table>
  );
  const tabElement = screen.getByText(/Tab 1/i);
  expect(tabElement).toBeInTheDocument();
});

test("renders when children with tab label is passed as prop", () => {
  render(
    <table>
      <tbody>
        <tr>
          <Tabs>
            <Tab label="tab text 1">Tab1</Tab>
            <Tab label="tab text 2">Tab2</Tab>
          </Tabs>
        </tr>
      </tbody>
    </table>
  );
  const tabElement = screen.getByText(/tab text 1/i);
  expect(tabElement).toBeInTheDocument();
});
