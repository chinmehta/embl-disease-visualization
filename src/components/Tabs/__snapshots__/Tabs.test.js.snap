// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when children is passed as prop - snapshot 1`] = `
<td
  className="border border-gray-400"
  colSpan="4"
>
  <div
    className="tab-container my-2 mx-5"
  >
    <div
      className="tab-list flex border-gray-400 border-b-2"
    >
      <span
        className=" px-4 cursor-pointer bg-embl-primary-blue text-white"
        onClick={[Function]}
      >
        tab text 1
      </span>
      <span
        className=" px-4 cursor-pointer bg-white text-embl-primary-blue"
        onClick={[Function]}
      >
        tab text 2
      </span>
    </div>
    <div
      className="flex h-80 justify-center items-center"
    >
      Tab1
    </div>
  </div>
</td>
`;

exports[`renders when no children is passed as prop 1`] = `
<td
  className="border border-gray-400"
  colSpan="4"
>
  <div
    className="tab-container my-2 mx-5"
  >
    <div
      className="tab-list flex border-gray-400 border-b-2"
    />
    <div
      className="flex h-80 justify-center items-center"
    />
  </div>
</td>
`;
